src/linter.ts

```diff
@@ -113,6 +114,7 @@ export class Linter {
 
     public lint(fileName: string, source: string, configuration: IConfigurationFile = DEFAULT_CONFIG): void {
         const sourceFile = this.getSourceFile(fileName, source);
+        if (configuration && isFileExcluded(fileName, configuration)) { return; }
```

Comment:
move this up one line. do the check first, don't even bother making `sourceFile`

Reply:
also please push the `configuration != null` check into `isFileExcluded` (the code already supports this, just need to make the arg optional)